api/bean/ConfigMapAndSecret.go

```diff
@@ -69,3 +71,81 @@ func (configSecretJson ConfigSecretJson) GetDereferencedSecrets() []ConfigSecret
 func (configSecretJson *ConfigSecretJson) SetReferencedSecrets(secrets []ConfigSecretMap) {
 	configSecretJson.Secrets = util.GetReferencedArray(secrets)
 }
+
+func GetDecodedDataForSecret(data string) (string, error) {
+	secretsJson := ConfigSecretRootJson{}
+	err := json.Unmarshal([]byte(data), &secretsJson)
+	if err != nil {
+		return "", err
+	}
+
+	for _, configData := range secretsJson.ConfigSecretJson.Secrets {
+		configData.Data = configData.GetDecodedData()
+	}
+
+	marshal, err := json.Marshal(secretsJson)
+	if err != nil {
+		return "", err
+	}
+	return string(marshal), nil
+}
+
+func (configSecretMap ConfigSecretMap) GetDecodedData() []byte {
```

Comment:
we can make these functions generic and pass the encoder/decoder as function

Reply:
done